### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by MayPyone**
- Added the validations
- Added spec for model

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  it "post counter shouldn't be valid" do
+    subject.posts_counter = nil
+    expect(subject).not_to be_valid
+  end
+
+  it 'post counter should be valid' do
+    subject.posts_counter = 5
+    expect(subject).to be_valid
+  end
```
Comment (thecodechaser) [inline spec/models/user_spec.rb:29]: - [ ] Please add a test for the `recent_posts` method in the user model as requested in the project requirements to add specs for all of your models' methods.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  it 'should be invalid for comment counters' do
+    subject.comment_counter = nil
+    expect(subject).to be_invalid
+  end
+
+  it 'Text should be the same' do
+    expect(subject.text).to eq('Hi')
+  end
```
Comment (thecodechaser) [inline spec/models/post_spec.rb]: - [ ] Please add tests for the `update_posts_counter`  and `recent_comments` methods in the post model as requested in the project requirements to add specs for all of your models' methods.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @MayPyone👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image

## To Highlight 🎉
- All tests are passed. ✔️ 
- Good job with validations. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @MayPyone, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**